shopinvader_sale_cart/models/sale_order.py

```diff
@@ -83,3 +83,35 @@ def _get_cart_line(self, product_id):
         return self.order_line.filtered(
             lambda l, product_id=product_id: l.product_id.id == product_id
         )[:1]
+
+    def _update_cart_lines_from_cart(self, cart):
+        self.ensure_one()
+        update_cmds = []
+        for cart_line in cart.order_line:
+            line = self._get_cart_line(cart_line.product_id.id)
+            if line:
+                new_qty = line.product_uom_qty + cart_line.product_uom_qty
+                vals = {"product_uom_qty": new_qty}
+                vals.update(line._play_onchanges_cart_line(vals))
+                cmd = (1, line.id, vals)
+            else:
+                vals = {
+                    "order_id": self.id,
+                    "product_id": cart_line.product_id.id,
+                    "product_uom_qty": cart_line.product_uom_qty,
+                }
+                vals.update(self.env["sale.order.line"]._play_onchanges_cart_line(vals))
+                cmd = (0, None, vals)
+            update_cmds.append(cmd)
+        self.write({"order_line": update_cmds})
+
+    def _transfer_cart(self, partner_id):
+        """
+        Transfer the current cart to a given partner
```

Comment:
Maybe expand this docstring. Something like this?
```suggestion
        Copy the lines of this cart to the current cart of a given partner. Merge lines of identical products.
        If the partner does not have a current cart yet, create one.
        This cart is not deleted. It is the responsibility of the caller to remove it if desired.
```

Reply:
>  This cart is not deleted. It is the responsibility of the caller to remove it if desired.

this part is very helpful :wink:

Reply:
maybe a `delete` option could be useful?

Reply:
Or another method `_transfer_and_unlink_cart`

Reply:
But in the end it's simply a matter of adding an `unlink` line at the call site, so a different method or a delete flag may not ease the life of the caller that much.

Reply:
Thanks for the input.

I like to keep it simple, I think this is not necessary to add a delete option or a `_transfer_and_unlink_cart()` method. As @sbidoul said, this is just a matter of adding an `unlink` after the call.

I have added the suggested docstring.

Reply:
I'm fine w/ the proposed solution, however... I don't really see the point for keeping this cart.
Especially in the ctx of the anon partner you MUST delete the cart.
As this meth is called "transfer cart" where do you see a use for keeping the old cart?

Reply:
I don't know, you tell me :joy: I thought you had a use case for that, as you started this discussion with this comment:

> Why deleting the cart?
> And what about the UUID?
> If you delete it you should keep they UUID for ext reference.

For me personally, the only use I have of this method is in the context of anonymous partners.  But I like the idea to keep this method atomic and let people do what they want with the cart.